src/datamodel/builder/UMessage.cpp

```diff
@@ -181,65 +181,65 @@ UMessageBuilder& UMessageBuilder::withCommStatus(v1::UCode code) {
 }
 
 UMessageBuilder& UMessageBuilder::withPayloadFormat(v1::UPayloadFormat format) {
-    if ((format < v1::UPayloadFormat_MIN) ||
-        (format > v1::UPayloadFormat_MAX)) {
-        throw std::out_of_range("Payload format value is out of range");
-    }
-
-    attributes_.set_payload_format(format);
+    setPayloadFormat(format);
 
     return *this;
 }
 
 v1::UMessage UMessageBuilder::build() const {
     v1::UMessage message;
-    if ((attributes_.payload_format() >= v1::UPayloadFormat_MIN) &&
-        (attributes_.payload_format() <= v1::UPayloadFormat_MAX)) {
+    if (expectedPayloadFormat_.has_value()) {
 
-        throw UnexpectedFormat("payload format did not match the one set with withPayloadFormat()");
+        throw UnexpectedFormat("Tried to build with no payload when a payload format has been set using withPayloadFormat()");
     }
 
-    message.set_allocated_attributes(new v1::UAttributes(attributes_));
+    *message.mutable_attributes() = attributes_;
 
     return message;
 }
 
 v1::UMessage UMessageBuilder::build(builder::Payload&& payload) const {
     v1::UMessage message;
 
-    message.set_allocated_attributes(new v1::UAttributes(attributes_));
+    *message.mutable_attributes() = attributes_;
 
     return message;
 }
 
 UMessageBuilder::UMessageBuilder(v1::UMessageType msgType,
                 v1::UUri&& source,
-                v1::UUri&& sink,
-                v1::UUID&& request_id) {
+                std::optional<v1::UUri>&& sink,
+                std::optional<v1::UUID>&& request_id) {
 
-    auto uuidBuilder = UuidBuilder::getBuilder();
-    auto uuid = uuidBuilder.build();
 
-    attributes_.set_allocated_id(new v1::UUID(uuid));
+    *attributes_.mutable_id() = UuidBuilder::getBuilder().build();
```

Comment:
Done, but this required removal of const qualifier at the end of the function because now thiis function needs to modify the state of the object it is called on.

Reply:
It shouldn't need to modify the state of the object. It can set the value in the message's copy of attributes after the copy is made. So instead of `*attributes_.mutable_id()...` you would do `*(message.mutable_attributes()->mutable_id())...`

Reply:
Makes sense, done